### PR TITLE
Wayland protocol update and keyboard inhibit support

### PR DESCRIPTION
**Pull request by akallabeth**
* Updates `xdg-shell` to stable API
* Updates `fullscreen-shell` to `fullscreen-shell-unstable-v1` API
* Add `keyboard-shortcuts-inhibit-unstable-v1` support to inhibit local keyboard shortcuts (`-grab-keyboard` to disable) Currently no release key implemented, only mouse can release again.
* Add `xdg-decorations` support to enable server side decorations if available
* Add `kde-decorations` support to enable server side decorations if available (fallback option)

### CODE REVIEW

**Comment by hardening** [comment]
Looks perfectly fine.

File: uwac/libuwac/uwac-window.c [hunk 1]
```diff
@@ -155,10 +157,11 @@ static void xdg_handle_configure(void* data, struct xdg_surface* surface,
 	}
 
 ack:
-	xdg_surface_ack_configure(surface, serial);
+	return;
 }
```
Comment (hardening) [inline uwac/libuwac/uwac-window.c]: lonely return ? (unneeded here)

**Comment by hardening** [comment]
@akallabeth remove the erroneous return and I'll merge this PR

File: uwac/libuwac/uwac-window.c (hunk 1 above)
Comment (akallabeth) [reply uwac/libuwac/uwac-window.c]: @hardening necessary for the label to jump to, did not want to refactor the whole function (but then again it is not that large)

Comment (akallabeth) [reply uwac/libuwac/uwac-window.c]: ok, done :)

**Pull request merged**